### PR TITLE
DAOS-4174 control: improve dmg storage prepare output

### PR DESCRIPTION
**Pull request by tanabarr**
Bring command output in line with storage scan and
format. Group and tabulate command results.

- improve connect client API consistency
- add prepare and scan unit test coverage
- remove unnecessary Stringer implementations
- convert types implicitly to isolate proto structs from dmg
- make prepare and scan output consistent between daos_server and dmg

Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>

### CODE REVIEW

File: src/control/client/storage.go [hunk 1]
```diff
@@ -63,10 +63,88 @@ func storagePrepareRequest(mc Control, req interface{}, ch chan ClientResult) {
...
+			c.setPrepareErr(cNvmePrepare, cScmPrepare, addr,
+				fmt.Errorf(msgBadType, &ctlpb.StoragePrepareResp{}, res.Value))
+			continue
+		}
+
+		if resp.GetNvme() == nil || resp.GetScm() == nil {
+			c.setPrepareErr(cNvmePrepare, cScmPrepare, addr,
+				fmt.Errorf("malformed response, missing submessage %+v", resp))
+			continue
+		}
```
Comment (kjacque) [inline src/control/client/storage.go]: What does the response look like if there's no NVMe storage in the system?

I'm assuming the case of no SCM is impossible (short of some kind of internal error) if we're running DAOS.

File: src/control/client/types.go [hunk 2]
```diff
@@ -297,6 +297,34 @@ func (rm ResultStateMap) String() string {
...
+type ScmPrepareResult struct {
+	Namespaces storage.ScmNamespaces
+	Err        error
+}
+
+// ScmPrepareMap maps ScmModulePrepareResult structs to the addresses
+// of remote servers identified by an address string.
+type ScmPrepareResults map[string]*ScmPrepareResult
+
+type NvmePrepareResult struct {
```
Comment (kjacque) [inline src/control/client/types.go]: nit - public type needs comment

File: src/control/cmd/dmg/storage.go [hunk 3]
```diff
@@ -286,3 +290,58 @@ func groupFormatResults(results client.StorageFormatResults, summary bool) (grou
...
+		}
+
+		sres := resp.Scm[srv]
+		switch {
+		case sres.Err != nil:
+			fmt.Fprintf(buf, "SCM Error: %s\n", sres.Err)
+		case len(sres.Namespaces) > 0:
+			fmt.Fprintf(buf, "%s\n", scmNsScanTable(sres.Namespaces))
+		default:
+			fmt.Fprintf(buf, "SCM ???\n")
```
Comment (kjacque) [inline src/control/cmd/dmg/storage.go]: I think it would be better to say something like "SCM Unknown State" or something like that

**Comment by mjmac** [comment]
@tanabarr: I'm going to close this PR on the assumption that the work has been completed elsewhere. We're trying to clean up outstanding review requests.

**Comment by tanabarr** [comment]
I believe that the improvements that have been made elsewhere are sufficient to close this, thanks Mike

**Pull request closed without merging**